### DISCUSSION TITLE
Question about towncrier

### DISCUSSION THREAD

**Discussion by MarkusPiotrowski** [beeware/toga#1727] [Q&A] [answered]
If I understood correctly, the fragments in `toga/changes` must be named with the PR #. However, when doing a PR, I can not be sure about the PR's number, can I? Is there a sort of 'blanco' identifier, which can be used?

**Comment by freakboy3742** [comment, marked as answer]
There's no "blank" identifier - you must provide a number. However:

1. Where possible, you should be using the *Issue* number, not the PR number. That way, the release notes will include links to the defined bugs/features that have been added, rather than a selection of random PRs
2. When there isn't an existing issue, you can create the PR in 2 passes - submit a PR *without* a change note - this will fail CI (fairly quickly, because the Towncrier check is one of the first checks performed), but it will also assign the PR number. You can then push an update to the PR, adding a change note with the assigned PR number.

**Reply by MarkusPiotrowski** [reply]
Should this information been added  to the 'How to contribute code' section of the docs?

**Reply by freakboy3742** [reply]
It certainly sounds like a good addition to the contribution guide to me.
